### DISCUSSION TITLE
Как использовать эту программу?

### DISCUSSION THREAD

**Discussion by RUSapache** [hufrea/byedpi#193] [Q&A] [answered]
Вот я скачал zip архив на Linux устройство, распаковал. А дальше что? Какой файл запускать чтобы активировать?

**Comment by EMWD** [comment]
Если ты скачал исходный код, то сначала его нужно скомпилировать. 
Если скачал уже скомпилированную программу, то нужно запускать её в терминале с правильными флагами, в готовом билде файл будет называться на подобии "ciadpi", "ciadpi-x86-64" и .т.д.

**Reply by RUSapache** [reply]
Скачал через зелёную кнопку Code (Download ZIP), в любом случае как мне выполнять указанные действия?

**Reply by chemizt** [reply]
@RUSapache если тебе нужно, чтобы "просто заработало" - скачивать исходники не имеет смысла, проще скачать готовый бинарь под свою архитектуру [отсюда](https://github.com/hufrea/byedpi/releases/tag/v0.14.1). После распаковки запускай ciadpi-x86_64 (если у тебя x86_64) как любую другую программу, примеры использования и описание параметров есть на главной странице

**Reply by RUSapache** [reply]
@chemizt Меня вполне устроят и исходники, я к ним инструкцию по применению не могу найти. И с готовыми тоже вообще ничего не понятно. Как понять какой файл под какую архитектуру? Я не вижу списка. При попытке их запустить ничего не происходит. Примеры тоже не понятные, там есть код и параметры, куда этот код вводить?

**Reply by Siberpone** [reply, marked as answer]
@RUSapache Ну если у тебя линух на обычном ПК, то нужен вот этот:
https://github.com/hufrea/byedpi/releases/download/v0.14.1/byedpi-14.1-x86_64.tar.gz

Качаешь, распаковываешь, запускаешь с рабочими параметрами. Параметры подбираются коллективно методом научного тыка (тут есть несколько тем, посвященных этому вопросу). Текущий (вроде бы) рабочий вариант: `-s1 -q1 -f-1 -S -b661 -As`. Т. е. полная команда в терминале будет выглядеть как-то так:
```
./ciadpi-x86_64 -s1 -q1 -f-1 -S -b661 -As
```
Эта команда запустит у тебя локальный прокси сервер, на который нужно перенаправлять проги (браузер, например), которые хочешь разблокать. Адрес у него будет `socks5://127.0.0.1:1080`. Соответственно, Хром/Хромоподобные бровзеры, например, можно перенаправить через терминал так:
```
chrome --proxy-server="socks5://127.0.0.1:1080"
```

**Reply by RUSapache** [reply]
@Siberpone У меня Steam Deck, браузер Firefox. После первой команды выдал "Bind: Address in use", после второй "Bash: chrome: command not found"

**Reply by Siberpone** [reply]
@RUSapache
> После первой команды выдал "Bind: Address in use"

Вероятно, прога осталась запущенной от предыдущих попыток. Нужно прибить её через таск менеджер или через терминал: `killall ciadpi`



> после второй "Bash: chrome: command not found"

Не, ну так само собой если у тебя Хром не установлен, то это не сработает :) Я ж чисто для примера привел. В этих ваших Фуррифоксах прокся через настройки задается. Открываешь настройки, забиваешь в поиск "proxy", тыкаешь на найденную строчку и в открывшемся окошке забиваешь все как на скриншоте (**внимание:** при посещении всяких скуфуслуг и прочих особо духоскрепных сайтов проксю, возможно, придется отрубать, ибо могут отфутболивать):
![scr-2024-10-12-002159](https://github.com/user-attachments/assets/97b60e9b-5a89-4777-b8a2-1d021da446ea)

**Reply by RUSapache** [reply]
Ага, после этого Youtube стал открывать домашнюю страницу и страницу с видео, но загружать видео так и не хочет. Видимо сейчас нужно опытным путём выбирать настройки.
Edit: После смены 1 цифры в коде видео Youtube начал работать исправно. Теперь желательно бы способ создания ярлыка для запуска терминала с кодом
